tests/json-fixtures/test_transactions.py

```diff
@@ -40,43 +57,91 @@
+    """
+    # TODO: This was copied from state tests. Make more DRY.
+    for fixture_path, fixture_key in all_fixtures:
+        fixture = load_fixture(fixture_path, fixture_key)
+        for fixture_fork, fork_states in fixture.items():
+            if (fixture_fork in ForkName.__members__):
+                yield fixture_path, fixture_key, fixture_fork
+
+
 def pytest_generate_tests(metafunc):
     generate_fixture_tests(
         metafunc=metafunc,
         base_fixture_path=BASE_FIXTURE_PATH,
+        preprocess_fn=expand_fixtures_forks,
     )
 
 
 @pytest.fixture
 def fixture(fixture_data):
-    fixture_path, fixture_key = fixture_data
+    fixture_path, fixture_key, fixture_fork = fixture_data
     fixture = load_fixture(
         fixture_path,
         fixture_key,
-        normalize_transactiontest_fixture,
+        normalize_transactiontest_fixture(fork=fixture_fork),
     )
+
     return fixture
 
+@pytest.fixture
+def fixture_transaction_class(fixture_data):
+    _, _, fork_name = fixture_data
+    if fork_name == ForkName.Frontier.name:
+        return FrontierTransaction
+    elif fork_name == ForkName.Homestead.name:
+        return HomesteadTransaction
+    elif fork_name == ForkName.EIP150.name:
+        return HomesteadTransaction
+    elif fork_name == ForkName.EIP158.name:
+        return SpuriousDragonTransaction
+    elif fork_name == ForkName.Byzantium.name:
+        return ByzantiumTransaction
+    elif fork_name == ForkName.Sharding.name:
+        pytest.skip("Sharding Transaction class has not been implemented")
+    elif fork_name == ForkName.Constantinople.name:
+        pytest.skip("Constantinople Transaction class has not been implemented")
+    elif fork_name == ForkName.Metropolis.name:
+        pytest.skip("Metropolis Transaction class has not been implemented")
+    else:
+        raise ValueError("Unknown Fork Name: {0}".format(fork_name))
+
 
-def test_transaction_fixtures(fixture):
-    header = BlockHeader(1, fixture['blocknumber'], 100)
-    chain = MainnetChain(ChainDB(get_db_backend()), header=header)
-    vm = chain.get_vm()
-    TransactionClass = vm.get_transaction_class()
+def test_transaction_fixtures(fixture, fixture_transaction_class):
+
+    #TODO: Clarify the following:
+    # It seems that previously the transaction class was determined by
+    # the block number. With the new fixture structure there's no block
+    # number and the test was altered to directly run against the specified
+    # transaction class from each provided fork. This means, we aren't implicitly
+    # testing anymore if we actually pick the correct transaction class based on the
+    # block number. Is this covered somewhere else?
+    TransactionClass = fixture_transaction_class
 
     if 'sender' in fixture:
         transaction = rlp.decode(fixture['rlp'], sedes=TransactionClass)
-        expected = normalize_signed_transaction(fixture['transaction'])
-
-        assert transaction.nonce == expected['nonce']
```

Comment:
@pipermerriam as I wrote in the PR description. The new fixture data doesn't have the `transaction` data anymore. Or maybe I didn't understand how to put the pieces together?

Reply:
Yep, I was thinking that basically we are just validating the `rlp` with the new tests. That said, there are still a couple of exceptions that we do need to catch and assert correctly (e.g. invalid RLP cases).

I must admit, I'm not entirely sure about the role of the `hash` in the fixtures. Does an existing `hash` just indicate that a transaction is valid whereas the absence of a `hash` in a transaction means it is invalid?

I also think that we do want to keep this part, no?

```
try:
    txn.validate()
except ValidationError:
    return
```